StreetHawk/Classes/Core/Publish/SHApp.m

```diff
@@ -1105,6 +1107,55 @@ - (void)applicationDidBecomeActiveNotificationHandler:(NSNotification *)notifica
     [[NSNotificationCenter defaultCenter] postNotificationName:@"SH_PushBridge_Smart_Notification" object:nil];
 }
 
+/**
+ check push permission and set tag sh_push_denied to current datetime if permission is denied, or delete the tag if permission change from NO to YES
+ */
+- (void)checkPushPermission
+{
+    if([self currentInstall].suid == nil){
+        return;
+    }
+    
+    if([[NSUserDefaults standardUserDefaults] objectForKey:@"registerForRemoteNotificationOccurred"] == nil){
```

Comment:
Some problems:
1. if customer deny permission when app first launch, this will stay false and below code never execute. 
2. the push notification should be handled in Notification Module, while this is Core Module. 
3. I read the ticket's description https://streethawk.atlassian.net/browse/IOS-1266. There is flag "revoked" in install/update to indicate notification is on or off. One solution is server can process "revoked" flag to add/remove tag; even if you want to do it in SDK, please check the code logic of "revoked".